### PR TITLE
[boost-fiber] Enable x64-osx

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?
  Enables x64-osx support for the boost-fiber port.
  In turn, this allows requiring vcpkg-ci-boost to pass for x64-osx.

  Disclaimer: I'm unaware of actual particularities of macOS support in this port. I see that people build it outside vcpkg, and I didn't find a reason for disabling it in the port history or upstream.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  Enabled x64-osx. Require vcpkg-ci-boost:x64-osx to pass.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  yes

### CODE REVIEW

**Comment by dg0yt** [comment]
> *This is a new experimental fast check for PR issues. Please let us know if this bot is helpful!*
> ...
> ~~~
> warning: there are uncommitted changes for ...


Hm, there can't be uncommitted changes because I can only push what is committed...

**Comment by Osyotr** [comment]
When touching boost-* ports, please also update `scripts/boost/generate-ports.ps1`.

File: scripts/boost/generate-ports.ps1 [hunk 1]
```diff
@@ -45,7 +45,7 @@ $portData = @{
     };
     "boost-beast"            = @{ "supports" = "!emscripten" };
     "boost-fiber"            = @{
-        "supports" = "!osx & !uwp & !arm & !emscripten";
```
Comment (Osyotr) [inline scripts/boost/generate-ports.ps1:48]: Also bump version in `$portVersions` above:
```posh
$portVersions = @{
    #e.g. "boost-asio" = 1;
    "boost-build" = 1;
    "boost-fiber" = 1;
    "boost-modular-build-helper" = 1;
}
```

Comment (dg0yt) [reply scripts/boost/generate-ports.ps1:48]: Wow, so much redundancy, and no hint in the generated file. I need to find a machine with Powershell.

Comment (Osyotr) [reply scripts/boost/generate-ports.ps1:48]: Agreed. I can add comment to generated manifest files in https://github.com/microsoft/vcpkg/pull/26817

**Review by Adela0814 — APPROVED** [review]

**Pull request merged**